tests/destinations/test_path_utils.py

```diff
@@ -1,46 +1,122 @@
+from typing import List, Tuple
+
+import pendulum
 import pytest
 
-from dlt.destinations import path_utils
+from dlt.common import logger
+from dlt.common.storages import LoadStorage
+from dlt.common.storages.load_package import ParsedLoadJobFileName
+
+from dlt.destinations.path_utils import (
+    check_layout,
+    create_path,
+    get_table_prefix_layout,
+)
+
 from dlt.destinations.exceptions import InvalidFilesystemLayout, CantExtractTablePrefix
+from tests.common.storages.utils import start_loading_file, load_storage
 
 
-def test_layout_validity() -> None:
-    path_utils.check_layout("{table_name}")
-    path_utils.check_layout("{schema_name}/{table_name}/{load_id}.{file_id}.{ext}")
-    with pytest.raises(InvalidFilesystemLayout) as exc:
-        path_utils.check_layout("{other_ph}.{table_name}")
-    assert exc.value.invalid_placeholders == ["other_ph"]
+TestLoad = Tuple[str, ParsedLoadJobFileName]
+EXTRA_PLACEHOLDERS = {"type": "one-for-all", "vm": "beam", "module": "__MODULE__"}
+ALL_LAYOUTS = (  # type: ignore
+    # Usual layouts
+    ("{schema_name}/{table_name}/{load_id}.{file_id}.{ext}", True, []),
+    ("{schema_name}.{table_name}.{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}88{load_id}-u-{file_id}.{ext}", True, []),
+    # Extra layouts
+    ("{table_name}/{curr_date}/{load_id}.{file_id}.{ext}{timestamp}", True, []),
+    ("{table_name}/{YYYY}-{MM}-{DD}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{YYYY}-{MMM}-{D}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{DD}/{HH}/{m}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{D}/{HH}/{mm}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{timestamp}/{load_id}.{file_id}.{ext}", True, []),
+    (
```

Comment:
this is the only case with extra placeholders in the block you marked with # Extra placeholders, clean it up